pkg/nextgen/distdoc/graph.go

```diff
@@ -0,0 +1,150 @@
+package distdoc
+
+import (
+	"container/list"
+	"errors"
+	"github.com/nuts-foundation/nuts-network/pkg/model"
+	"sort"
+)
+
+var errRootAlreadyExists = errors.New("root document already exists")
+
+// DAG is a directed acyclic graph consisting of nodes (documents) referring to preceding nodes.
+type DAG struct {
+	// root contains the root node of the DAG
+	root  *node
+	nodes map[string]*node
```

Comment:
idea to define a type for the hash as string?

Reply:
Actually, I changed Hash from []byte to [20]byte to allow for using as map key but I forgot to change it...